## *Check Current Directories and Changing Directories* ##
What current directory are we in?


In [1]:
from pathlib import Path
import os
cwd = os.getcwd()
print(cwd)

/home/rogertu/WikiData


Current directory is in /home/rogertu/WikiData on BlacksBeach server in WD_Test virtual environment.  The directory we want to map to is /scheduled-bots/scheduled_bots/ontology/obographs_DO.py

In [2]:
# maps filename variable to a specific location
filename = Path("scheduled-bots/scheduled_bots/ontology/obographs_DO.py")
print("This is the file name ", filename.name)
print("This is the suffix ", filename.suffix)


This is the file name  obographs_DO.py
This is the suffix  .py


## *Run obographs_DO.py* ##
Does script run? Added print statement at end and commented out final lines.

In [3]:
#run file
%run -i 'scheduled-bots/scheduled_bots/ontology/obographs_DO.py'

ModuleNotFoundError: No module named 'wikidataintegrator'

## *Why are modules not loading?* ##

Why are we not importing packages? Is it because we don't have an init?

In [16]:
pwd

'/home/rogertu/WikiData'

In [18]:
import scheduled-bots

SyntaxError: invalid syntax (<ipython-input-18-959bdd1fa264>, line 1)

In [19]:
pip


Usage:   
  /home/rogertu/WikiData/WD_Test/bin/python3 -m pip <command> [options]

Commands:
  install                     Install packages.
  download                    Download packages.
  uninstall                   Uninstall packages.
  freeze                      Output installed packages in requirements format.
  list                        List installed packages.
  show                        Show information about installed packages.
  check                       Verify installed packages have compatible dependencies.
  config                      Manage local and global configuration.
  search                      Search PyPI for packages.
  wheel                       Build wheels from your requirements.
  hash                        Compute hashes of package archives.
  completion                  A helper command used for command completion.
  help                        Show help for commands.

General Options:
  -h, --help                  Show help.
  --isolated      

In [21]:
import importlib

In [22]:
importlib.import_module("scheduled-bots")

<module 'scheduled-bots' (namespace)>

In [25]:
import util

ModuleNotFoundError: No module named 'util'

In [20]:
pip list

Package                Version  Location                             
---------------------- -------- -------------------------------------
antlr4-python3-runtime 4.7.2    
asn1crypto             0.24.0   
attrs                  19.1.0   
backcall               0.1.0    
bcrypt                 3.1.6    
beautifulsoup4         4.7.1    
biothings-client       0.2.0    
bleach                 3.1.0    
bs4                    0.0.1    
cachetools             3.1.0    
Cerberus               1.2      
certifi                2019.3.9 
cffi                   1.12.2   
CFGraph                0.2.1    
chardet                3.0.4    
Click                  7.0      
cryptography           2.6.1    
DateTime               4.3      
decorator              4.4.0    
defusedxml             0.5.0    
entrypoints            0.3      
idna                   2.8      
ipykernel              5.1.0    
ipython                7.3.0    
ipython-genutils       0.2.0    
ipywidgets             7.4.2    
is

In [4]:
%%bash
cd /home/rogertu/WikiData/scheduled-bots/scheduled_bots/
ls

cgi
civic
disease_ontology
drugs
faers
fda
genedisease
gene_disease_phenotype
geneprotein
iedb
__init__.py
interpro
inxight
local.py
logger
notes.md
ontology
pbb_tracker
phenotypes
query_tester
reactome
scripts
shextest
talk_jenkins
trigger
utils.py
variant
wikipathways


Definitely have an "__init__"

In [5]:
import scheduled_bots

ModuleNotFoundError: No module named 'scheduled_bots'

In [6]:
import utils

ModuleNotFoundError: No module named 'utils'

In [7]:
import scheduled_bots.utils

ModuleNotFoundError: No module named 'scheduled_bots'

Seems like the module isn't showing up... Does python know it exists?

In [8]:
help("modules")


Please wait a moment while I gather a list of all available modules...



/home/rogertu/WikiData/WD_Test/lib/python3.6/site-packages/IPython/kernel/__init__.py:13: ShimWarning: The `IPython.kernel` package has been deprecated since IPython 4.0.You should import from ipykernel or jupyter_client instead.
  "You should import from ipykernel or jupyter_client instead.", ShimWarning)
/home/rogertu/WikiData/WD_Test/lib/python3.6/site-packages/nbconvert/exporters/exporter_locator.py:28: DeprecationWarning: `nbconvert.exporters.exporter_locator` is deprecated in favor of `nbconvert.exporters.base` since nbconvert 5.0.
  DeprecationWarning)


CFGraph             bcrypt              jupyter             reprlib
DateTime            bdb                 jupyter_client      requests
IPython             binascii            jupyter_console     requests_oauthlib
SPARQLWrapper       binhex              jupyter_core        resource
ShExJSG             biothings_client    jwt                 rlcompleter
__future__          bisect              keyword             rmagic
_ast                bleach              lib2to3             runpy
_asyncio            bs4                 linecache           sched
_bisect             bson                locale              secrets
_blake2             builtins            logging             select
_bootlocale         bz2                 lxml                selectors
_bz2                cProfile            lzma                send2trash
_cffi_backend       cachetools          macpath             setuptools
_codecs             calendar            macurl2path         shelve
_codecs_cn          cerberus   

/usr/lib/python3.6/pkgutil.py:107: VisibleDeprecationWarning: zmq.eventloop.minitornado is deprecated in pyzmq 14.0 and will be removed.
    Install tornado itself to use zmq with the tornado IOLoop.
    
  yield from walk_packages(path, info.name+'.', onerror)


In [9]:
%%bash
which python
python --version

/home/rogertu/WikiData/WD_Test/bin/python
Python 3.6.7


## *Troubleshooting scheduled_bots (utils module)* ##
```from scheduled_bots import utils``` scheduled_bots exists under scheduled-bots. I can't import `scheduled-bots` because it doesn't fit proper python naming convention for import. `utils` exists under`scheduled_bots`. `scheduled_bots` should be recognized by system as it has an `__init__`.  Installed `setup.py` with `pip install -e .`. Module now hangs at obographs... why?

In [10]:
%%bash
pwd

/home/rogertu/WikiData


In [11]:
%%bash
python3 -m scheduled-bots.scheduled_bots.ontology.obographs_DO

Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/rogertu/WikiData/scheduled-bots/scheduled_bots/ontology/obographs_DO.py", line 4, in <module>
    from obographs import Graph, Node
ModuleNotFoundError: No module named 'obographs'


CalledProcessError: Command 'b'python3 -m scheduled-bots.scheduled_bots.ontology.obographs_DO\n'' returned non-zero exit status 1.

In [12]:
scheduled-bots.scheduled_bots.ontology.obographs_DO

NameError: name 'scheduled' is not defined

In [14]:
from WikiData import util

ModuleNotFoundError: No module named 'WikiData'

In [15]:
import WikiData

ModuleNotFoundError: No module named 'WikiData'

## *Need to install WikiDataIntegrator.....*##

In [1]:
%%bash
pwd
python3 scheduled-bots/scheduled_bots/ontology/obographs_DO.py

/home/rogertu/WikiData


usage: obographs_DO.py [-h] [--local] json_path
obographs_DO.py: error: the following arguments are required: json_path


CalledProcessError: Command 'b'pwd\npython3 scheduled-bots/scheduled_bots/ontology/obographs_DO.py\n'' returned non-zero exit status 2.

## *Add JSON path* ##

In [2]:
%%bash
pwd
python3 scheduled-bots/scheduled_bots/ontology/obographs_DO.py .

/home/rogertu/WikiData
https://www.wikidata.org/w/api.php
Successfully logged in as Torogertu
Done running obographs_DO


## *Changed JSON file to Shortened JSON file* ##

In [1]:
%%bash
pwd
python3 scheduled-bots/scheduled_bots/ontology/obographs_DO.py .

/home/rogertu/WikiData
https://www.wikidata.org/w/api.php
Successfully logged in as Torogertu
Done running obographs_DO


## *Uncomment last two lines and run...* ##

In [3]:
%%bash
pwd
python3 scheduled-bots/scheduled_bots/ontology/obographs_DO.py ./sampleDOID.json

/home/rogertu/WikiData
https://www.wikidata.org/w/api.php
Successfully logged in as Torogertu


Traceback (most recent call last):
  File "scheduled-bots/scheduled_bots/ontology/obographs_DO.py", line 94, in <module>
    g = DOGraph(args.json_path, mediawiki_api_url=mediawiki_api_url, sparql_endpoint_url=sparql_endpoint_url)
  File "/home/rogertu/WikiData/scheduled-bots/scheduled_bots/ontology/obographs.py", line 327, in __init__
    self.load_graph()
  File "/home/rogertu/WikiData/scheduled-bots/scheduled_bots/ontology/obographs.py", line 344, in load_graph
    d = json.load(f)
  File "/usr/lib/python3.6/json/__init__.py", line 299, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "/usr/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.6/json/decoder.py", line 355, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting

CalledProcessError: Command 'b'pwd\npython3 scheduled-bots/scheduled_bots/ontology/obographs_DO.py ./sampleDOID.json\n'' returned non-zero exit status 1.

## *Seems that self-selected `.json` file doesn't work* ##

Generated a new `.json` file called `doidtest.json`. This file only has 2 entries.  We deleted all nodes and edges and only added these entries in nodes and edges for (DOID_13146) RO_0001025 and edges for (DOID_0060002) RO_0002200.  

In [2]:
%%bash
pwd
python3 scheduled-bots/scheduled_bots/ontology/obographs_DO.py ./doidtest.json

/home/rogertu/WikiData
https://www.wikidata.org/w/api.php
Successfully logged in as Torogertu
P699 not found in fastrun
P2888 not found in fastrun
P2892 not found in fastrun
P1748 not found in fastrun
P1692 not found in fastrun
P4229 not found in fastrun
skipping 641 edges where the subject is a node that is being skipped
loading: P1554
P927 not found in fastrun
the following should be checked and deleted: {}
{'P4229', 'P927', 'P1692', 'P699', 'P1748', 'P2892'}
P4229 not found in fastrun
P927 not found in fastrun
P1692 not found in fastrun
P699 not found in fastrun
P1748 not found in fastrun
P2892 not found in fastrun
-----
Q5398631
[('P699', 'DOID:13146'), ('P1748', 'C27027'), ('P2892', 'C0239295')]
[]
Done running obographs_DO


creating edges: 100%|██████████| 1/1 [00:02<00:00,  2.19s/it]
Waiting for endpoint update: 2019-03-27 22:40:38 / 2019-03-27 22:40:11.188542 : : 3it [01:00, 15.48s/it]


## *No edits were made to DOID_13146 (esophageal candidiasis)* ##
Nothing was removed, nothing was added. For DOID_0060002, because there was no node entry, the program couldn't find a 'subject' to add disease relationships to.


On WikiData the corresponding property ID's are...
> P699 - Disease Ontology ID <br>
P2888 - exact match <br>
P2892 - UMLS CUI <br>
P1748 - NCI Thesaurus ID <br>
P1692 - ICD-9-CM (identifier in the ICD adaption assigning diagnostic and procedure codes <br>
P4229 - ICD-10-CM (classification of medical condition) <br>

Is the missing P-ID's in the entry because of missing 'node' for DOID_0060002?



On WikiData the corresponding property ID's are...
> P699 - Disease Ontology ID <br>
**P927 - anatomical location** <br>
P2892 - UMLS CUI <br>
P1748 - NCI Thesaurus ID <br>
P1692 - ICD-9-CM (identifier in the ICD adaption assigning diagnostic and procedure codes <br>
P4229 - ICD-10-CM (classification of medical condition) <br>

Where anatomical location was the only property that wasn't found in `fast run`

> `Q5398631
[('P699', 'DOID:13146'), ('P1748', 'C27027'), ('P2892', 'C0239295')]
[]`

Where (Disease ontology ID, esophageal candidiasis), (NCI Metathesaurus, the Code), (CUI ID, the Code)]. Both NCI Metathesaursus and CUI ID point to the same item. (makes sense cause its a synonym).

## *What happens when we include the missing node (DOID_0060002)* ##


In [4]:
%%bash
python3 scheduled-bots/scheduled_bots/ontology/obographs_DO.py ./doidtest2.json

https://www.wikidata.org/w/api.php
Successfully logged in as Torogertu
P699 not found in fastrun
P2888 not found in fastrun
P1692 not found in fastrun
P1748 not found in fastrun
P4229 not found in fastrun
P2892 not found in fastrun
skipping 641 edges where the subject is a node that is being skipped
loading: P3841
None;None;None;no qids found for: http://purl.obolibrary.org/obo/HP_0004431;None
loading: P1554
P927 not found in fastrun
the following should be checked and deleted: {}
{'P6532', 'P927', 'P4229', 'P2892', 'P1692', 'P1748', 'P699'}
P6532 not found in fastrun
P927 not found in fastrun
P4229 not found in fastrun
P2892 not found in fastrun
P1692 not found in fastrun
P1748 not found in fastrun
P699 not found in fastrun
-----
Q5398631
[('P2892', 'C0239295'), ('P1748', 'C27027'), ('P699', 'DOID:13146')]
[]
Done running obographs_DO


creating edges: 100%|██████████| 2/2 [00:02<00:00,  1.26it/s]
Waiting for endpoint update: 2019-03-27 23:25:55 / 2019-03-27 23:25:28.800880 : : 3it [01:00, 15.49s/it]


## *No decrease in output* ##

New `P3841`, `None;None;None; no qids found for http://purl.obolibrary.org/obo/HP_0004431`, and `P6532`

> P3841 - Human Phenotype Ontology ID <br>
P6532 - has phenotype <br>
hyperlink - corresponds to "Complement deficiency - an immunodeficiency defined by the absent or suboptimal functioning of one of the complement system proteins."

However, a QID definitely exists on WikiData `https://www.wikidata.org/wiki/Q5156409`.  The purl link on WikiData refers to DOID_626.  Both refer to Complement deficiency, could be a misannotation as HP points to Human phenotype ontology and DO points to Human Disease ontology.

No Q18553584 output in this and previous case, and no edit to the WikiData page.

## *Checked to see if Props were in the __init__* ##

Missing elements were found in `__init__.py`.

In [26]:
%%bash
cd /home/rogertu/WikiData/scheduled-bots/scheduled_bots/
items=('P6532 P927 P4229 P2892 P1692 P1748 P699')

for element in $items; do
    if grep -q $element __init__.py; then
        echo "found $element in __init__.py"
    else
        echo "missing $element"
    fi
done

found P6532 in __init__.py
found P927 in __init__.py
found P4229 in __init__.py
found P2892 in __init__.py
found P1692 in __init__.py
found P1748 in __init__.py
found P699 in __init__.py


## *Checked where `fastrun` is found* ##
Is `fastrun` something editable in my directory or is it a file in WDI

In [28]:
%%bash
cd /home/rogertu/WikiData/scheduled-bots/scheduled_bots/ontology/
grep -r "fastrun" ./

Binary file ./__pycache__/obographs.cpython-35.pyc matches
Binary file ./__pycache__/obographs.cpython-36.pyc matches
./obo_importer.py:    parser.add_argument('--fastrun', dest='fastrun', action='store_true')
./obo_importer.py:    parser.add_argument('--no-fastrun', dest='fastrun', action='store_false')
./obo_importer.py:    parser.set_defaults(fastrun=True)
./obo_importer.py:    fast_run = args.fastrun
./obographs.py:    def _get_fastrun_container(self):
./obographs.py:        # get a fastrun container
./obographs.py:            print("fastrun container not found. not removing deprecated statements")
./obographs.py:        frc = self._get_fastrun_container()
./obographs.py:        frc = self._get_fastrun_container()


All fingers point to WDI core in wikidataintegrator.  However, didn't have time to figure out what the issue was.  Based on this testing, Andrew and I want to move forward with a different test case for `has phenotype` for test case complement deficiency.  Next steps are to build a new test case.